app/models/setting.rb

```diff
@@ -1,11 +1,23 @@
 class Setting < ActiveRecord::Base
   belongs_to :user
 
-  def self.bart_max_distance
-    find_by(name: "bart_max_distance").value
-  end
+  REGISTERED_SETTINGS = [
+    :bart_max_distance,
+    :email_body,
+    :transit_max_time,
+    :walk_score_minimum,
+    :transit_score_minimum,
+    :bike_score_minimum,
+    :personal_crime_grade_minimum,
+    :property_crime_grade_minimum,
+  ]
 
-  def self.email_body
-    find_by(name: "email_body").value
+  def self.register_setting_getters
+    self.class.instance_eval do
+      REGISTERED_SETTINGS.each do |setting_name|
```

Comment:
If you move this out of the method and into the class body directly, does it still work?